project-base/app/src/DataFixtures/Demo/OrderWithdrawalRequestDataFixture.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataFixtures\Demo;
+
+use App\Model\Order\Order;
+use DateTime;
+use Doctrine\Common\DataFixtures\DependentFixtureInterface;
+use Doctrine\Persistence\ObjectManager;
+use Override;
+use Shopsys\FrameworkBundle\Component\DataFixture\AbstractReferenceFixture;
+use Shopsys\FrameworkBundle\Component\DataFixture\Exception\PersistentReferenceNotFoundException;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Translation\Translator;
+use Shopsys\FrameworkBundle\Model\Order\Withdrawal\WithdrawalRequestDataFactory;
+use Shopsys\FrameworkBundle\Model\Order\Withdrawal\WithdrawalRequestFacade;
+
+class OrderWithdrawalRequestDataFixture extends AbstractReferenceFixture implements DependentFixtureInterface
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Withdrawal\WithdrawalRequestFacade $withdrawalRequestFacade
+     * @param \Shopsys\FrameworkBundle\Model\Order\Withdrawal\WithdrawalRequestDataFactory $withdrawalRequestDataFactory
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly WithdrawalRequestFacade $withdrawalRequestFacade,
+        protected readonly WithdrawalRequestDataFactory $withdrawalRequestDataFactory,
+        protected readonly Domain $domain,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function load(ObjectManager $manager): void
+    {
+        foreach ($this->domain->getAll() as $domainConfig) {
+            try {
+                $orderReference = $this->getReferenceForDomain(OrderDataFixture::ORDER_WITH_WITHDRAWAL_REQUEST, $domainConfig->getId(), Order::class);
+
+                $withdrawalRequestData = $this->withdrawalRequestDataFactory->createFromArray([
+                    'firstName' => 'Adam',
+                    'lastName' => 'Bořič',
+                    'telephone' => '+420755496328',
+                    'email' => 'no-reply@shopsys.com',
+                    'note' => t('Product does not match description, I want to return the entire order.', domain: Translator::DATA_FIXTURES_TRANSLATION_DOMAIN, locale: $domainConfig->getLocale()),
+                ]);
+                $withdrawalRequestData->requestedAt = (new DateTime('now -1 day'))->setTime(14, 30, 15);
+
+                $this->withdrawalRequestFacade->createOnly($orderReference, $withdrawalRequestData);
+            } catch (PersistentReferenceNotFoundException) {
+                continue;
```

Comment:
Is this safe? Couldn't it just hide an errors and making debugging harder?

Reply:
yeah, the thing is (was) that we are creating distinct data for the 2nd domain in the `OrderDataFixture` and that was missing the reference. I fixed that by adding the reference even for the 2nd domain. IMHO it is better now, thanks for pointing this out.

eca3bfe98c